addons/barcodes/static/src/js/barcode_form_view.js

```diff
@@ -38,14 +38,21 @@ FormController.include({
         };
 
         this.barcodeMutex = new concurrency.Mutex();
+    },
+
+    /**
+     * @override
+     */
+    on_attach_callback: function () {
+        this._super.apply(this, arguments);
         this._barcodeStartListening();
     },
     /**
      * @override
      */
-    destroy: function () {
+    on_detach_callback: function () {
+        this._super.apply(this, arguments);
         this._barcodeStopListening();
-        this._super();
```

Comment:
do you remember why we need this patch? and it should be in another commit

Reply:
Apparently, its the proper way enable/disable the barcode listener (According to the framework JS team). In our case we had to do this because of concurrency problems with the barcode listener when opening the inventory client action on top of the picking client action.